### DISCUSSION TITLE
DNS server auto selection

### DISCUSSION THREAD

**Discussion by mrx23dot** [DNSCrypt/dnscrypt-proxy#1919] [Q&A] [answered]
Does dnscrypt-proxy select fastest DNS server based on **actual (no-error response)** and not simply on ping?

**Comment by jedisct1** [comment, marked as answer]
Actual queries are sent.

The `ping` command is not good at measuring DNS latency, if the server responds to it at all.
`ping` also wouldn't be able to measure anything via relays.
